src/state/actions/dispatchANTInteraction.ts

```diff
@@ -136,6 +146,30 @@ export default async function dispatchANTInteraction({
         }
         break;
       }
+      case ANT_INTERACTION_TYPES.REASSIGN_NAME: {
+        let newAntProcessId = payload.newAntProcessId;
+        if (!newAntProcessId) {
+          dispatchSigningMessage('Spawning new ANT, please wait... 1/2');
+          newAntProcessId = await spawnANT({
+            ao,
+            signer: createAoSigner(signer),
+            state: payload.previousState,
+            luaCodeTxId: payload.luaCodeTxId,
+          });
+        }
+        dispatchSigningMessage(
+          'Reassigning ArNS name, please wait... ' +
+            (payload.newAntProcessId ? '' : '2/2'),
+        );
+        // for UX so the user sees the signing message
+        await sleep(2000);
+        result = await antProcess.reassignName({
+          name: payload.name,
+          arioProcessId: payload.arioProcessId,
+          antProcessId: newAntProcessId,
+        });
+        break;
+      }
```

Comment:
could explain further on the comment here. Also, `await spawnANT` should not end until that message is already signed

Reply:
ah, I get it -- its to see the `    'Reassigning ArNS name, please wait... ' +` message

Reply:
Exactly yeah - noticed the reassign was a bit quick sometimes and it didn't even give the message time to rerender.